packages/aws-cdk/lib/api/aws-auth/aws-sdk-inifile.ts

```diff
@@ -54,39 +58,78 @@ export class PatchedSharedIniFileCredentials extends AWS.SharedIniFileCredential
     var externalId = roleProfile.external_id;
     var mfaSerial = roleProfile.mfa_serial;
     var sourceProfileName = roleProfile.source_profile;
+    var sourceCredentials = roleProfile.credential_source;
 
-    if (!sourceProfileName) {
+    if (sourceProfileName && sourceCredentials) {
       throw (AWS as any).util.error(
-        new Error('source_profile is not set using profile ' + this.profile),
+        new Error('source_profile and credential_source are both configured in profile ' + this.profile + '. please choose one or the other.'),
         { code: 'SharedIniFileCredentialsProviderFailure' },
       );
     }
 
-    var sourceProfileExistanceTest = creds[sourceProfileName];
-
-    if (typeof sourceProfileExistanceTest !== 'object') {
+    if (!sourceProfileName && !sourceCredentials) {
       throw (AWS as any).util.error(
-        new Error('source_profile ' + sourceProfileName + ' using profile '
-          + this.profile + ' does not exist'),
+        new Error('neither source_profile nor credential_source configured in profile ' + this.profile + '. please configure one or the other.'),
```

Comment:
The error message should contain something like "if not using static credentials".

It's not *mandatory* to use assume-role credentials, and the user seeing this message doesn't necessarily know the maze of `if`s we went through to get to this place in the code.

Reply:
Not sure i follow, if we get here, doesn't that mean the user had explicitly set a profile during the invocation? I don't like "if not" messages, i'd rather indicate what the user *is* doing, which is using a profile, and thats already mentioned.

No?

Reply:
Rephrased the error a bit, let me know if thats better.